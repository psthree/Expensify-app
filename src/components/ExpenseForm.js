import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

const now = moment();
console.log(now.format('MMM Do, YYYY'));

export default class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      description: props.expense ? props.expense.description : '',
      note: props.expense ? props.expense.note : '',
      amount: props.expense ? (props.expense.amount / 100).toString() : '',
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      calendarFocused: false,
      error: ''
    };
  }

  onDescriptionChange = e => {
    const description = e.target.value;
    this.setState(() => ({ description }));
  };
  onNoteChange = e => {
    const note = e.target.value;
    this.setState(() => ({ note }));
  };
  onAmountChange = e => {
    const amount = e.target.value;
    //must be a number with no more that 2 numbers after the decimal
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }));
    }
  };
  onDateChange = createdAt => {
    if (createdAt) {
      this.setState(() => ({ createdAt }));
    }
  };
  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }));
  };
  onSubmit = e => {
    e.preventDefault();
    //validate if desc or amount is empty
    if (!this.state.description || !this.state.amount) {
      //set error stat equal to 'Please provide Description and amount'
      this.setState(() => ({
        error: 'Please provide Description and amount'
      }));
    } else {
      //clear error
      // console.log('Submitted');
      this.setState(() => ({
        error: ''
      }));

      //onSubmit is added to prpos when component is used in AddExpense.js
      this.props.onSubmit({
        description: this.state.description,
        //amount needs to be convert to a number (base 10) is gets here as a string
        //and get rid of decimal place(* 100)
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };
  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form action="" onSubmit={this.onSubmit}>
          <input
            type="text"
            placeholder="Description"
            autoFocus
            value={this.state.description}
            onChange={this.onDescriptionChange}
          />
          <input
            type="text"
            placeholder="Amount"
            value={this.state.amount}
            onChange={this.onAmountChange}
          />
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={() => false}
          />
          <textarea
            name=""
            id=""
            placeholder="Add a note for your expense (optional)"
            onChange={this.onNoteChange}
          />
          <button>Add Expense</button>
        </form>
      </div>
    );
  }
}
