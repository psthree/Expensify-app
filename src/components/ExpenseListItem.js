//export a stateless functional component
//description, amount created at
import React from 'react';
import { Link } from 'react-router-dom';



const ExpenseListItem = ({ id, description, amount, createdAt }) => {
    console.log(`test ${id}`);
    return (
        <div>
            <Link to={`/edit/${id}`} >
                <h3>{description}</h3>
            </Link>
            <p>{amount} - {createdAt}</p>

        </div>


    )
}

export default (ExpenseListItem);