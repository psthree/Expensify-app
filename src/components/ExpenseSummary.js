import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const formatedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div>
      <h1>
        Viewing {expenseCount} {expenseWord} totaling {formatedExpensesTotal}
      </h1>
    </div>
  );
};

export default ExpensesSummary;
