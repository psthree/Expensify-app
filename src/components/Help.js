import React from 'react';

const Help = () => (
    <div>This is my HELP component</div>
);

export default Help;