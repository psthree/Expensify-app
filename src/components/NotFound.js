import React from 'react';
import { Link } from 'react-router-dom';

const NotFound = () => (
    <div>
        404 Error (page not found)
        <Link to="/" >Dashboard</Link>
    </div>
);

export default NotFound;