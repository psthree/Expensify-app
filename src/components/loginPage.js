import React from 'react';
//import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

//destructuring the startLogins
export const LoginPage = ({ startLogin }) => (
  <div>
    <button onClick={startLogin}>Login</button>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin())
});

//export default LoginPage;
//undefined is mapStateToProps
export default connect(
  undefined,
  mapDispatchToProps
)(LoginPage);
