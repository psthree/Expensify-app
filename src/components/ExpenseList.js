//stateless functional component
import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import getVisibleExpenses from '../selectors/expenses';

// this is ONLY exported for testing, the app uses the default export
export const ExpenseList = props => (
  <div>
    {props.expenses.length === 0 ? (
      <p>No expenses</p>
    ) : (
      props.expenses.map(expense => {
        return <ExpenseListItem key={expense.id} {...expense} />;
      })
    )}
  </div>
);

const mapStateToProps = state => {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters)
  };
};

export default connect(mapStateToProps)(ExpenseList);
