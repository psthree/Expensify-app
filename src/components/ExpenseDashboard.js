import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummary from './ExpenseSummary';

const ExpenseDashboard = () => (
  <div>
    This is my dashboard component
    <ExpenseSummary expenseCount={1} expensesTotal={239} />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboard;
