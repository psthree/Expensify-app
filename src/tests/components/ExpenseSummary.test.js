import React from 'react';
import { shallow } from 'enzyme';
import numeral from 'numeral';
import { ExpensesSummary } from '../../components/ExpenseSummary';

test('Should correctly render expensesSummary with one expense', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={1} expensesTotal={239} />
  );
  expect(wrapper).toMatchSnapshot();
});

test('Should correctly render expensesSummary with multiple expense', () => {
  const wrapper = shallow(
    <ExpensesSummary expenseCount={15} expensesTotal={23998897} />
  );
  expect(wrapper).toMatchSnapshot();
});

// u updates snapshot
// npm test-- - u
