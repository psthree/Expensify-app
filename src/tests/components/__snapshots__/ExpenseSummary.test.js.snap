// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render expensesSummary with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    15
     
    expenses
     totaling 
    $239,988.97
  </h1>
</div>
`;

exports[`Should correctly render expensesSummary with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totaling 
    $2.39
  </h1>
</div>
`;
